src/nupic/regions/AnomalyLikelihoodRegion.py

```diff
@@ -0,0 +1,148 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""Implementation of region for computing anomaly likelihoods."""
+
+from nupic.algorithms.anomaly_likelihood import AnomalyLikelihood
+from nupic.bindings.regions.PyRegion import PyRegion
+
+
+class AnomalyLikelihoodRegion(PyRegion):
+  """Region for computing the anomaly likelihoods."""
+
+
+  @classmethod
+  def getSpec(cls):
+    return {
+      "description": ("Region that computes anomaly likelihoods for \
+                       temporal memory."),
+      "singleNodeOnly": True,
+      "inputs": {
+        "rawAnomalyScore": {
+          "description": "The anomaly score whose \
+                          likelihood is to be computed",
+          "dataType": "Real32",
+          "count": 1,
+          "required": True,
+          "isDefaultInput": False
+        },
+        "value": {
+          "description": "The input metric value",
+          "dataType": "Real32",
+          "count": 1,
+          "required": True,
+          "isDefaultInput": False
+        },
+      },
+      "outputs": {
+        "anomalyLikelihood": {
+          "description": "The resultant anomaly likelihood",
+          "dataType": "Real32",
+          "count": 1,
+          "isDefaultOutput": True,
+        },
+      },
+      "parameters": {
+        "claLearningPeriod": {
```

Comment:
No particular rationale for that term - it was put in years ago. Keep in mind backward compatibility though - various products are using the anomaly likelihood module directly.

In this class we shouldn't use the term CLA.